addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -946,15 +946,30 @@ var SnippetsMenu = Widget.extend({
         // again when the mouse moves
         this.$document.on('keydown.snippets_menu', () => {
             this.snippetEditors.forEach(editor => {
-                editor.toggleTextEdition(true);
+                editor.toggleOverlayHidden(true);
             });
         });
         this.$document.on('mousemove.snippets_menu, mousedown.snippets_menu', () => {
             this.snippetEditors.forEach(editor => {
-                editor.toggleTextEdition(false);
+                editor.toggleOverlayHidden(false);
             });
         });
 
+        // Hide the active overlay when scrolling.
+        // Show it again and recompute all the overlays after the scroll.
+        this.$document.on('scroll.snippets_menu', () => {
```

Comment:
You have to debounce/throttle that function, you can't do stuff at each scroll event...